### PR TITLE
#105 deb package distribution and systemd support

### LINKED ISSUE #105: Investigate: run project as a systemd service in debian based linux distros

### PR DESCRIPTION
**Pull request by Krillsson**
closes #105

### CODE REVIEW

**Comment by Krillsson** [comment]
@Droid-MAX I'm working on distributing a .deb file and systemd service definition inspired by your fork. Got it up and running on latest ubuntu-server but would be awesome to have a second pair of eyes on this as working with systemd is new to me.

File: build.gradle [hunk 1]
```diff
@@ -22,11 +23,71 @@ plugins {
...
+    }
+
+    from('config') {
+        fileType CONFIG | NOREPLACE
+        into '/opt/sys-api/config'
+    }
+
+    from('src/deb/opt/sys-api/data') {
+        createDirectoryEntry = true
+        fileMode 0777
```
Comment (Krillsson) [inline build.gradle:86]: Not necessary since we chmod the whole directory tree in the postinst script

File: build.gradle [hunk 2]
```diff
@@ -22,11 +23,71 @@ plugins {
...
+        into '/opt/sys-api/lib'
+    }
+
+    from('config') {
+        fileType CONFIG | NOREPLACE
+        into '/opt/sys-api/config'
+    }
+
+    from('src/deb/opt/sys-api/data') {
+        createDirectoryEntry = true
```
Comment (Krillsson) [inline build.gradle:85]: Make sure this doesn't overwrite files

**Comment by Droid-MAX** [comment]
https://github.com/Krillsson/sys-API/blob/e29c9d943d68ed8b209706fbf3fdb9d7ebd0a663/src%2Fdeb%2Flib%2Fsystemd%2Fsystem%2Fsys-api.service#L11

Maybe we can remove this line of code, forget what the exit status code 23 corresponds to.

**Comment by Krillsson** [comment]
Thanks for the checkup 🙏🏻

**Comment by Krillsson** [comment]
Sample .deb file here: https://github.com/Krillsson/sys-API/actions/runs/11087049073#artifacts

**Pull request merged**